tests/iostability/test_iostability_happy_path.py

```diff
@@ -256,17 +257,23 @@ def test_iteration_write_read_partial_delete(self):
                     resp = NearFullStorage.delete_workload(workload_info_list, self.s3userinfo,
                                                            delete_percent_per_iter)
                     assert_utils.assert_true(resp[0], resp[1])
-
+                    write_per = write_per - delete_percent_per_iter
                 else:
                     self.log.warning("No buckets available to perform read,validate,delete"
                                      " operations %s", workload_info_list)
             else:
                 self.log.info("Write percentage(%s) exceeding the max cluster capacity(%s)",
                               write_per, max_cluster_capacity_percent)
-                self.log.info("Deleting all the written data.")
-                resp = NearFullStorage.delete_workload(workload_info_list, self.s3userinfo, 100)
-                assert_utils.assert_true(resp[0], resp[1])
-                self.log.info("Deletion completed.")
+                if len(workload_info_list) > 0:
+                    self.log.info("Deleting all the written data.")
+                    resp = NearFullStorage.delete_workload(workload_info_list, self.s3userinfo, 100)
+                    assert_utils.assert_true(resp[0], resp[1])
+                    self.log.info("Deletion completed.")
+                    write_per = 0
+                else:
+                    self.log.warning("No buckets available to perform read,validate,delete"
+                                     " operations %s", workload_info_list)
+                    assert False, "Specified Max capacity already attained."
```

Comment:
please check if we want to fail test on this condition or we can just log warning and break from loop?

Reply:
We want to fail the test here. Because if the system is already filled 95%(max_percentage) , there can two conditions. 
1. Write were performed as part of test, in which case workload_info_list should be > 0. 
2.  If prefilled setup is provided, No IOs will be made by test as we don't delete data which was not written as part of test and hence marking pass can be misleading. 
Hence asserting here for point 2.

Reply:
ok